src/RequestHandler.php

```diff
@@ -211,6 +245,11 @@ public function slaveClosed()
             return sprintf('<info>Worker %d took abnormal %.3f seconds for handling a connection</info>', $this->slave->getPort(), $took);
         });
 
+        // Return a "502 Bad Gateway" response if the response was empty
+        if($this->lastOutgoingData == '') {
```

Comment:
Coding style

Reply:
This will capture no response from slave and relay partial responses, like client breaks during headers? Good idea, I was gonna make it unneccessarily complex.

Reply:
The whole response is buffered afaik so it should catch all cases where the script did not return a valid HTTP response object (e.g. if it called exit, if it crashed etc). To test it I added the following lines in a Symfony controller:
```
        header('Location: http://www.google.com');
        echo 'test';
        flush();
        ob_flush();
        exit();
```
And PHP-PM returned the 502 response as expected. This logic might need to be reconsidered later when PHP-PM supports streamed responses, but for now I didn't want to actually process the response considering it could be some big PDF or other :)